install/source/dependencies/index.markdown

```diff
@@ -71,15 +67,21 @@ Next manually add a package.xml as used in the ROS release wrapper for OMPL (mod
 
     wget https://raw.githubusercontent.com/ros-gbp/ompl-release/debian/kinetic/xenial/ompl/package.xml
 
+Lastly, because of overlay issues you have to change the order of include variables in **MoveIt's** build system.
+Sadly, this cannot be changed in the upstream project, because it would break the default use case.
+To understand this limitation see this [Github issue](https://github.com/ros-planning/moveit/issues/169#issuecomment-242849008):
+
+    Edit moveit/moveit_planners/ompl/CMakeLists.txt
+    Locate the include_directories directive and exchange the two variables ${OMPL_INCLUDE_DIRS} and ${catkin_INCLUDE_DIRS},
+    so that ${OMPL_INCLUDE_DIRS} is added before ${catkin_INCLUDE_DIRS}.
+
+
 Now you should be able to build using regular ``catkin build``
 
 ## Flexible Collision Library (FCL)
 
 FCL is MoveIt!'s default collision checker, their official install instructions can be found [here](https://github.com/flexible-collision-library/fcl).
 
-To install from source within catkin, first remove any binary installed verions:
-
-    sudo apt-get -qq remove libfcl0 libfcl-dev ros-kinetic-octomap
```

Comment:
this should not be removed and is unrelated to this PR

Reply:
No it is not unrelated and it should be removed from the instructions, unless I'm mistaken.

- uninstalling octomap has nothing to do with building FCL from source, especially when there is no instructions to build octomap from source too.
- the point of this pull-request is to *leave system libraries installed* and instead overlay them correctly.

Reply:
Your fix in this PR concerns OMPL, and doesn't address FCL. You should remove this unrelated change so we can merge the important parts of this PR

We've built FCL from source very recently and https://github.com/ros-planning/moveit.ros.org/pull/174 takes those learnings and shares them with everyone. Have you built FCL from source recently?